userPayeeEndorsement.go

```diff
@@ -256,6 +235,42 @@ func (upe *UserPayeeEndorsement) Validate() error {
 				Value: upe.InternalControlSequenceNumber, Msg: err.Error()}
 		}
 	}
+	if upe.EndorsementIndicatorField() != "" {
+		if err := upe.isEndorsementIndicator(upe.EndorsementIndicator); err != nil {
+			return &FieldError{FieldName: "EndorsementIndicator",
+				Value: upe.EndorsementIndicatorField(), Msg: err.Error()}
+		}
+	}
+
+	if upe.UserField != "" {
+		if err := upe.isAlphanumericSpecial(upe.UserField); err != nil {
+			return &FieldError{FieldName: "UserField", Value: upe.UserField, Msg: err.Error()}
+		}
+	}
+
+	if err := upe.validateNameNumberFields(); err != nil {
+		return err
+	}
+	return nil
+
+}
+
+func (upe *UserPayeeEndorsement) validateNameNumberFields() error {
+	if upe.PayeeName != "" {
```

Comment:
Are all these fields allowed to be empty? Does there need to be at least one non-empty? 

Do BankRoutingNumber and BankAccountNumber either both need to be specified or both blank?

Reply:
PayeeName, .BankRoutingNumber, BankAccountNumber, OperatorName, OperatorNumber,
ManagerName, ManagerNumber, EquipmentNumber are all conditional fields.  They are not mandatory.

RE BankRoutingNUmber - From the X9 spec it "should" be associated with the BankAccountNumber

A number that identifies the institution or the organization where the item is being deposited. (note: should be the routing number associated with the Bank Account Number (Field 11)).

I'm not clear of an instance of where it wouldn't but since this is custom to specific clearing arrangements I didn't make the association.

Reply:
I'm not sure exactly how we would enforce it.